packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,267 @@
+- The converged `Checkbox` should also support `circular` checkboxes which both `v8` and `v0` do not support.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Checkbox label="Example Checkbox"/>
+
+<Checkbox label={<>Example Checkbox with <a href="https://www.microsoft.com">link</a></>} />
+
+<Checkbox circular size="large" label="Circular Checkbox" />
+
+<Checkbox label="Controlled Checkbox" onChange={onChangeFunction} />
+
+<Checkbox indeterminate label="Indeterminate Checkbox" />
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- A Checkbox has two size variants: `medium (default)` and `large`.
+- A Checkbox has three appearance variants: `disabled`, `unchecked`, `checked`, and `indeterminate`.
+- A Checkbox has a `circular` variant.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+### Checkbox Props
+
+```tsx
+/**
+ * Checkbox Props
+ */
+export interface CheckboxProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /**
+   * Label to be rendered with the checkbox.
+   */
+  label?: ShorthandProps<LabelProps>;
+
+  /**
+   * Icon to be displayed when the checkbox is in the checked state.
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Disabled state of the checkbox.
+   */
+  disabled?: boolean;
+
+  /**
+   * A checkbox can be rendered with a circular shape.
+   */
+  circular?: boolean;
+
+  /**
+   * A checkbox's state can be controlled.
+   * @defaultvalue false
+   */
+  checked?: 'indeterminate' | boolean;
+
+  /**
+   * Whether the checkbox should be rendered as checked by default.
+   */
+  defaultChecked?: 'indeterminate' | boolean;
+
+  /**
+   * Checkbox supports two different checkbox sizes.
+   * @defaultvalue 'medium'
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Determines whether the label should be positioned before (start) or after (end) the checkbox.
+   * @defaultvalue 'end'
+   */
+  labelPosition?: 'start' | 'end';
+
+  /**
+   * ID of the root element that wraps the checkbox and label.
+   */
+  rootId?: string;
+
+  /**
+   * ID of the <input/> element that represents the checkbox.
+   */
+  id?: string;
+
+  /**
+   * Callback to be called when the checked state value changes.
+   */
+  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: CheckboxOnChangeData) => void;
+}
+
+/**
+ * Data for the onChange event for checkbox.
+ */
+export interface CheckboxOnChangeData {
+  checked?: 'indeterminate' | boolean;
+}
+```
+
+Some feedback needed for props are:
+
+- `checked` will work similar to `aria-checked` and have `true`, `false`, and `indeterminate` options.
+- A `rootId` is needed since the input and label will share an id and `rootId` will be used for the root element.
```

Comment:
If this have not been discussed on a Tech Sync yet I suggest to do it 🙄 We need to have a pattern for such cases as `Input` component will have the same issue.

Reply:
Elizabeth and I are working on an RFC to figure out which pattern to follow. Right now this is the pattern but there's another option of using the `input` element as root and having a wrapper slot.